### ISSUE TITLE
cdsapi: ERA5 total precipitation accumulation

### ISSUE THREAD

**Issue by Aquila96** [HFAiLab/FourCastNet#23]
In `data_factory/download_era5.py`, it downloads 6-hour total precipitation by specifying 6 hour intevals.  

But after reviewing the data and reading the [ERA5 cdsapi demo](https://confluence.ecmwf.int/display/CKB/ERA5%3A+How+to+calculate+daily+total+precipitation),  It seems that the api only downloads 1-hour accumlated precipitation, for multi-hour accumulated precipitation, one have to download all the previous hours and perform accumulation afterwards.

**Comment by daniellucs2002** [comment]
Yeah, I agree with you. Also, in the GraphCast paper, the authors also mentioned that the precipitation data needs to be accumulated from the past six hours. So, have you tried to modify the code or figured out how to prepare the training data at least? (I plan to train the model from scratch if the provided weights file cannot be utilized)

**Comment by Aquila96** [comment]
> Yeah, I agree with you. Also, in the GraphCast paper, the authors also mentioned that the precipitation data needs to be accumulated from the past six hours. So, have you tried to modify the code or figured out how to prepare the training data at least? (I plan to train the model from scratch if the provided weights file cannot be utilized)

cdsapi is rather straight forward to use, you can refer to NVIDIA's download demo [here](https://github.com/NVlabs/FourCastNet/blob/master/copernicus/get_data_sfc_short_length.py) and [here](https://github.com/NVlabs/FourCastNet/blob/master/copernicus/get_data_pl_short_length.py). For precipitation specifically, I just fetch data from all 6 previous hours and add them together, similar to the approach in the ECMWF demo.

**Comment by daniellucs2002** [comment]
Thanks for answering! I remember you mentioned that the inconsistency in model architecture between huggingface checkpoint and this repository in another issue post. Have you addressed that problem and run the project on the GPU cluster?

**Comment by Aquila96** [comment]
> Thanks for answering! I remember you mentioned that the inconsistency in model architecture between huggingface checkpoint and this repository in another issue post. Have you addressed that problem and run the project on the GPU cluster?

No problem. Regarding the model weights, I wouldn't recommend to try to modify the source code to match the weights provided. 
For FourCastNet, we have compared results both from this version and the official version from NVIDIA, and the results from this version is worse, though I couldn't remember by how large of a margin.
For GraphCast, deepmind have since released their implementation and weights in jax, and the results are mostly on par with their claimed state-of-the-art performance. You can also read the implementation in modulus for the pytorch.
